<!DOCTYPE html>
<html lang="en-us">

<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="generator" content="Source Themes Academic 4.5.0">

  

  
  
  
  
  
    
    
    
  
  

  <meta name="author" content="Mohammed O.E Abdallah">

  
  
  
    
  
  <meta name="description" content="In this blog post I will be introducing cellbaseR R package, which makes use of the exhaustive RESTful Web service API that has been implemented for the Cellabase database. It enable researchers to query and obtain a wealth of biological information from a single database saving a lot of time. Another benefit is that researchers can easily make queries about different biological topics and link all this information together as all information is integrated">

  
  <link rel="alternate" hreflang="en-us" href="https://melsiddieg.github.io/post/simplifying-genomic-annotations-in-r/">

  


  
  
  
  <meta name="theme-color" content="#2962ff">
  

  
  
  
  
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.0/css/all.css" integrity="sha384-aOkxzJ5uQz7WBObEZcHvV5JvRW3TUc2rNPA7pe3AwnsUohiw1Vj2Rgx2KSOkF5+h" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" integrity="sha256-ygkqlh3CYSUri3LhQxzdcm0n1EQvH2Y+U5S2idbLtxs=" crossorigin="anonymous">

    
    
    
      
    
    
      
      
        
          <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/styles/github.min.css" crossorigin="anonymous" title="hl-light">
          <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/styles/dracula.min.css" crossorigin="anonymous" title="hl-dark" disabled>
        
      
    

    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.2.0/leaflet.css" integrity="sha512-M2wvCLH6DSRazYeZRIm1JnYyh22purTM+FDB5CsyxtQJYeKq83arPe5wgbNmcFXGqiSH2XR8dT/fJISVA1r/zQ==" crossorigin="anonymous">
    

    

  

  
  
  
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Montserrat:400,700%7CRoboto:400,400italic,700%7CRoboto+Mono&display=swap">
  

  
  
  
  <link rel="stylesheet" href="/css/academic.min.a3cda781a9b24f26fefd34a53547835b.css">

  

  




  


  

  <link rel="manifest" href="/index.webmanifest">
  <link rel="icon" type="image/png" href="/img/icon-32.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="https://melsiddieg.github.io/post/simplifying-genomic-annotations-in-r/">

  
  
  
  
    
    
  
  
  <meta property="twitter:card" content="summary">
  
  <meta property="twitter:site" content="@mohammedsiddieg">
  <meta property="twitter:creator" content="@mohammedsiddieg">
  
  <meta property="og:site_name" content="Academic">
  <meta property="og:url" content="https://melsiddieg.github.io/post/simplifying-genomic-annotations-in-r/">
  <meta property="og:title" content="Simplifying Genomic Annotations in R | Academic">
  <meta property="og:description" content="In this blog post I will be introducing cellbaseR R package, which makes use of the exhaustive RESTful Web service API that has been implemented for the Cellabase database. It enable researchers to query and obtain a wealth of biological information from a single database saving a lot of time. Another benefit is that researchers can easily make queries about different biological topics and link all this information together as all information is integrated"><meta property="og:image" content="https://melsiddieg.github.io/img/icon-192.png">
  <meta property="twitter:image" content="https://melsiddieg.github.io/img/icon-192.png"><meta property="og:locale" content="en-us">
  
    
      <meta property="article:published_time" content="2019-10-02T00:00:00&#43;00:00">
    
    <meta property="article:modified_time" content="2019-10-02T10:19:33&#43;02:00">
  

  


    






  






<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "https://melsiddieg.github.io/post/simplifying-genomic-annotations-in-r/"
  },
  "headline": "Simplifying Genomic Annotations in R",
  
  "datePublished": "2019-10-02T00:00:00Z",
  "dateModified": "2019-10-02T10:19:33+02:00",
  
  "author": {
    "@type": "Person",
    "name": "Mohammed O.E Abdallah"
  },
  
  "publisher": {
    "@type": "Organization",
    "name": "Academic",
    "logo": {
      "@type": "ImageObject",
      "url": "https://melsiddieg.github.io/img/icon-512.png"
    }
  },
  "description": "In this blog post I will be introducing cellbaseR R package, which makes use of the exhaustive RESTful Web service API that has been implemented for the Cellabase database. It enable researchers to query and obtain a wealth of biological information from a single database saving a lot of time. Another benefit is that researchers can easily make queries about different biological topics and link all this information together as all information is integrated"
}
</script>

  

  


  


  





  <title>Simplifying Genomic Annotations in R | Academic</title>

</head>

<body id="top" data-spy="scroll" data-offset="70" data-target="#TableOfContents" >

  <aside class="search-results" id="search">
  <div class="container">
    <section class="search-header">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" spellcheck="false" type="search">
        
      </div>

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>


  
<nav class="navbar navbar-light fixed-top navbar-expand-lg py-0 compensate-for-scrollbar" id="navbar-main">
  <div class="container">

    
      <a class="navbar-brand" href="/">Academic</a>
      
      <button type="button" class="navbar-toggler" data-toggle="collapse"
              data-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
        <span><i class="fas fa-bars"></i></span>
      </button>
      

    
    <div class="collapse navbar-collapse" id="navbar">

      
      
      <ul class="navbar-nav mr-auto">
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#about"><span>Home</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#posts"><span>Posts</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#projects"><span>Projects</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#publications"><span>Publications</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#contact"><span>Contact</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link " href="/files/MOE_CV.pdf"><span>CV</span></a>
        </li>

        
        

      
      </ul>
      <ul class="navbar-nav ml-auto">
      

        

        
        <li class="nav-item">
          <a class="nav-link js-search" href="#"><i class="fas fa-search" aria-hidden="true"></i></a>
        </li>
        

        

        
        <li class="nav-item">
          <a class="nav-link js-dark-toggle" href="#"><i class="fas fa-moon" aria-hidden="true"></i></a>
        </li>
        

      </ul>

    </div>
  </div>
</nav>


  <article class="article">

  












  

  
  
  
<div class="article-container pt-3">
  <h1>Simplifying Genomic Annotations in R</h1>

  

  
    



<div class="article-metadata">

  
  

  
  <span class="article-date">
    
    
      
          Last updated on
      
    
    Ocotober 10, 2109
  </span>
  

  

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    7 min read
  </span>
  

  
  
  

  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fas fa-folder mr-1"></i><a href="/categories/genomics/">Genomics</a>, <a href="/categories/variant-annotation/">variant annotation</a>, <a href="/categories/cellbaser/">cellbaseR</a>, <a href="/categories/bioconductor/">Bioconductor</a></span>
  

  
    
<div class="share-box" aria-hidden="true">
  <ul class="share">
    
      
      
      
        
      
      
      
      <li>
        <a href="https://twitter.com/intent/tweet?url=https://melsiddieg.github.io/post/simplifying-genomic-annotations-in-r/&amp;text=Simplifying%20Genomic%20Annotations%20in%20R" target="_blank" rel="noopener" class="share-btn-twitter">
          <i class="fab fa-twitter"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.facebook.com/sharer.php?u=https://melsiddieg.github.io/post/simplifying-genomic-annotations-in-r/&amp;t=Simplifying%20Genomic%20Annotations%20in%20R" target="_blank" rel="noopener" class="share-btn-facebook">
          <i class="fab fa-facebook-f"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="mailto:?subject=Simplifying%20Genomic%20Annotations%20in%20R&amp;body=https://melsiddieg.github.io/post/simplifying-genomic-annotations-in-r/" target="_blank" rel="noopener" class="share-btn-email">
          <i class="fas fa-envelope"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.linkedin.com/shareArticle?url=https://melsiddieg.github.io/post/simplifying-genomic-annotations-in-r/&amp;title=Simplifying%20Genomic%20Annotations%20in%20R" target="_blank" rel="noopener" class="share-btn-linkedin">
          <i class="fab fa-linkedin-in"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://web.whatsapp.com/send?text=Simplifying%20Genomic%20Annotations%20in%20R%20https://melsiddieg.github.io/post/simplifying-genomic-annotations-in-r/" target="_blank" rel="noopener" class="share-btn-whatsapp">
          <i class="fab fa-whatsapp"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://service.weibo.com/share/share.php?url=https://melsiddieg.github.io/post/simplifying-genomic-annotations-in-r/&amp;title=Simplifying%20Genomic%20Annotations%20in%20R" target="_blank" rel="noopener" class="share-btn-weibo">
          <i class="fab fa-weibo"></i>
        </a>
      </li>
    
  </ul>
</div>


  

</div>

    














  
</div>



  <div class="article-container">

    <div class="article-style">
      

<p>In this blog post I will be introducing cellbaseR R package, which  makes use of the exhaustive RESTful Web service API that has been implemented for the Cellabase       database. It enable researchers to query and obtain a wealth of biological information from a single database saving a lot of time. Another benefit is that researchers can easily make queries about different biological topics and link all this         information together as all information is integrated</p>

<pre><code class="language-r"># to get the default CellbaseR object (human data, from genome GRCh37)
library(cellbaseR)
# A default cellbaseR object is created as follows
cb &lt;- CellBaseR()
</code></pre>

<h2 id="the-cellbaser-methods">The CellbaseR Methods</h2>

<p>The cellbaseR package
provide many methods to query the cellbase webservices, they include:</p>

<ul>
<li>getGene<br /></li>
<li>getSnp</li>
<li>getProtein<br /></li>
<li>getTranscript</li>
<li>getRegion</li>
<li>getVariant</li>
<li>getClinical</li>
<li>getTf</li>
<li>getXref<br /></li>
</ul>

<p>In all cases the user is expected to provide the ids for the query and the
resource to be queried, in addition to the CellbaseQuery object they created.
For example, a query through the cbGene will look something like this</p>

<h3 id="getgene">getGene</h3>

<pre><code class="language-r">library(cellbaseR)
cb &lt;- CellBaseR()
genes &lt;- c(&quot;TP73&quot;,&quot;TET1&quot;)
res &lt;- getGene(object = cb, ids = genes, resource = &quot;info&quot;)
str(res,2)
</code></pre>

<pre><code>## 'data.frame':	2 obs. of  12 variables:
##  $ id         : chr  &quot;ENSG00000078900&quot; &quot;ENSG00000138336&quot;
##  $ name       : chr  &quot;TP73&quot; &quot;TET1&quot;
##  $ biotype    : chr  &quot;protein_coding&quot; &quot;protein_coding&quot;
##  $ status     : chr  &quot;KNOWN&quot; &quot;KNOWN&quot;
##  $ chromosome : chr  &quot;1&quot; &quot;10&quot;
##  $ start      : int  3569084 70320413
##  $ end        : int  3652765 70454239
##  $ strand     : chr  &quot;+&quot; &quot;+&quot;
##  $ source     : chr  &quot;Ensembl&quot; &quot;Ensembl&quot;
##  $ description: chr  &quot;tumor protein p73 [Source:HGNC Symbol;Acc:HGNC:12003]&quot; &quot;tet methylcytosine dioxygenase 1 [Source:HGNC Symbol;Acc:HGNC:29484]&quot;
##  $ transcripts:List of 2
##   ..$ :'data.frame':	14 obs. of  20 variables:
##   ..$ :'data.frame':	1 obs. of  20 variables:
##  $ annotation :'data.frame':	2 obs. of  2 variables:
##   ..$ expression:List of 2
##   ..$ diseases  :List of 2
</code></pre>

<pre><code class="language-r"># as you can see the res dataframe also contains a transcripts column 
# which is in fact a list column of nested dataframes, to get the
# trasnscripts data.frame for first gene
TET1_transcripts &lt;- res$transcripts[[1]]
str(TET1_transcripts,1)
</code></pre>

<pre><code>## 'data.frame':	14 obs. of  20 variables:
##  $ id                : chr  &quot;ENST00000378295&quot; &quot;ENST00000604074&quot; &quot;ENST00000354437&quot; &quot;ENST00000357733&quot; ...
##  $ name              : chr  &quot;TP73-001&quot; &quot;TP73-006&quot; &quot;TP73-002&quot; &quot;TP73-202&quot; ...
##  $ biotype           : chr  &quot;protein_coding&quot; &quot;protein_coding&quot; &quot;protein_coding&quot; &quot;protein_coding&quot; ...
##  $ status            : chr  &quot;KNOWN&quot; &quot;KNOWN&quot; &quot;KNOWN&quot; &quot;KNOWN&quot; ...
##  $ chromosome        : chr  &quot;1&quot; &quot;1&quot; &quot;1&quot; &quot;1&quot; ...
##  $ start             : int  3569084 3569084 3569129 3569129 3569129 3598930 3598930 3607236 3607236 3607236 ...
##  $ end               : int  3652765 3652765 3649856 3650467 3650467 3649643 3649643 3649856 3649856 3652765 ...
##  $ strand            : chr  &quot;+&quot; &quot;+&quot; &quot;+&quot; &quot;+&quot; ...
##  $ genomicCodingStart: int  3598930 3598930 3598930 3598930 3598930 3598930 3598930 3607470 3607470 3607470 ...
##  $ genomicCodingEnd  : int  3649643 3649326 3649326 3649643 3649643 3649643 3649643 3649326 3648119 3649643 ...
##  $ cdnaCodingStart   : int  156 156 111 111 111 1 1 235 235 235 ...
##  $ cdnaCodingEnd     : int  2066 1367 1610 1778 1733 1668 1623 1587 1515 1998 ...
##  $ cdsLength         : int  1910 1211 1499 1667 1622 1667 1622 1352 1280 1763 ...
##  $ proteinID         : chr  &quot;ENSP00000367545&quot; &quot;ENSP00000475143&quot; &quot;ENSP00000346423&quot; &quot;ENSP00000350366&quot; ...
##  $ proteinSequence   : chr  &quot;MAQSTATSPDGGTTFEHLWSSLEPDSTYFDLPQSSRGNNEVVGGTDSSMDVFHLEGMTTSVMAQFNLLSSTMDQMSSRAASASPYTPEHAASVPTHSPYAQPSSTFDTMSP&quot;| __truncated__ &quot;MAQSTATSPDGGTTFEHLWSSLEPDSTYFDLPQSSRGNNEVVGGTDSSMDVFHLEGMTTSVMAQFNLLSSTMDQMSSRAASASPYTPEHAASVPTHSPYAQPSSTFDTMSP&quot;| __truncated__ &quot;MAQSTATSPDGGTTFEHLWSSLEPDSTYFDLPQSSRGNNEVVGGTDSSMDVFHLEGMTTSVMAQFNLLSSTMDQMSSRAASASPYTPEHAASVPTHSPYAQPSSTFDTMSP&quot;| __truncated__ &quot;MAQSTATSPDGGTTFEHLWSSLEPDSTYFDLPQSSRGNNEVVGGTDSSMDVFHLEGMTTSVMAQFNLLSSTMDQMSSRAASASPYTPEHAASVPTHSPYAQPSSTFDTMSP&quot;| __truncated__ ...
##  $ cDnaSequence      : chr  &quot;TGCCTCCCCGCCCGCGCACCCGCCCGGAGGCTCGCGCGCCCGCGAAGGGGACGCAGCGAAACCGGGGCCCGCGCCAGGCCAGCCGGGACGGACGCCGATGCCCGGGGCTGC&quot;| __truncated__ &quot;TGCCTCCCCGCCCGCGCACCCGCCCGGAGGCTCGCGCGCCCGCGAAGGGGACGCAGCGAAACCGGGGCCCGCGCCAGGCCAGCCGGGACGGACGCCGATGCCCGGGGCTGC&quot;| __truncated__ &quot;AGGGGACGCAGCGAAACCGGGGCCCGCGCCAGGCCAGCCGGGACGGACGCCGATGCCCGGGGCTGCGACGGCTGCAGAGCGAGCTGCCCTCGGAGGCCGGCGTGGGGAAGA&quot;| __truncated__ &quot;AGGGGACGCAGCGAAACCGGGGCCCGCGCCAGGCCAGCCGGGACGGACGCCGATGCCCGGGGCTGCGACGGCTGCAGAGCGAGCTGCCCTCGGAGGCCGGCGTGGGGAAGA&quot;| __truncated__ ...
##  $ xrefs             :List of 14
##  $ tfbs              :List of 14
##  $ exons             :List of 14
##  $ annotationFlags   :List of 14
</code></pre>

<h3 id="getregion">getRegion</h3>

<pre><code class="language-r"># making a query through cbRegion to get all the clinically relevant variants 
# in a specific region
library(cellbaseR)
cb &lt;- CellBaseR()
res &lt;- getRegion(object=cb,ids=&quot;17:1000000-1005000&quot;,
resource=&quot;clinical&quot;)
# to get all conservation data in this region
res &lt;- getRegion(object=cb,ids=&quot;17:1000000-1005000&quot;,
resource=&quot;conservation&quot;)
#likewise to get all the regulatory data for the same region
res &lt;- getRegion(object=cb,ids=&quot;17:1000000-1005000&quot;, resource=&quot;regulatory&quot;)
str(res,1)
</code></pre>

<pre><code>## 'data.frame':	60 obs. of  12 variables:
##  $ chromosome  : chr  &quot;17&quot; &quot;17&quot; &quot;17&quot; &quot;17&quot; ...
##  $ source      : chr  &quot;ccat_histone&quot; &quot;SWEmbl_R0005_IDR&quot; &quot;SWEmbl_R0005_IDR&quot; &quot;SWEmbl_R0005_IDR&quot; ...
##  $ featureType : chr  &quot;H3K27_trimethylation_site&quot; &quot;H3K27_acylation_site&quot; &quot;H3K4_monomethylation_site&quot; &quot;H3K9_acetylation_site&quot; ...
##  $ start       : int  993140 997759 997760 998344 998400 999107 999880 997760 998344 998400 ...
##  $ end         : int  1004420 1000704 1002709 1000418 1000080 1000583 1000680 1002709 1000418 1000080 ...
##  $ score       : chr  &quot;.&quot; &quot;.&quot; &quot;.&quot; &quot;.&quot; ...
##  $ strand      : chr  &quot;.&quot; &quot;.&quot; &quot;.&quot; &quot;.&quot; ...
##  $ frame       : chr  &quot;.&quot; &quot;.&quot; &quot;.&quot; &quot;.&quot; ...
##  $ name        : chr  &quot;H3K27me3&quot; &quot;H3K27ac&quot; &quot;H3K4me1&quot; &quot;H3K9ac&quot; ...
##  $ featureClass: chr  &quot;Histone&quot; &quot;Histone&quot; &quot;Histone&quot; &quot;Histone&quot; ...
##  $ alias       : chr  &quot;K562_H3K27me3_ENCODE_YALE_ccat_histone&quot; &quot;Monocytes-CD14+_H3K27ac_ENCODE_Broad_SWEmbl_R0005_IDR&quot; &quot;Monocytes-CD14+_H3K4me1_ENCODE_Broad_SWEmbl_R0005_IDR&quot; &quot;Monocytes-CD14+_H3K9ac_ENCODE_Broad_SWEmbl_R0005_IDR&quot; ...
##  $ cellTypes   :List of 60
</code></pre>

<h3 id="getvariant">getVariant</h3>

<p>Getting annotation for a specific variant</p>

<pre><code class="language-r">library(cellbaseR)
cb &lt;- CellBaseR()
res2 &lt;- getVariant(object=cb, ids=&quot;1:169549811:A:G&quot;, resource=&quot;annotation&quot;)
# to get the data 
res2 &lt;- cbData(res2)
str(res2, 1)
</code></pre>

<p>A very powerfull feature of cellbaseR is ability to fetch a wealth of clinical
data, as well as abiliy to fiter these data by gene, phenotype, rs, etcâ€¦</p>

<h3 id="getclinical">getClinical</h3>

<pre><code class="language-r">library(cellbaseR)
cb &lt;- CellBaseR()
# First we have to specify aour param, we do that by creating an object of
# class CellbaseParam
cbparam &lt;- CellBaseParam(feature = &quot;TET1&quot;, assembly = &quot;GRCh38&quot;, limit = 50)
cbparam
</code></pre>

<pre><code>## |An object of class CellBaseParam
## |use this object to control what results are returned from the CellBaseR
##       methods
</code></pre>

<pre><code class="language-r"># Note that cbClinical does NOT require any Ids to be passed, only the param
# and of course the CellbaseQuery object
res &lt;- getClinical(object=cb,param=cbparam)
str(res,1)
</code></pre>

<pre><code>## 'data.frame':	50 obs. of  16 variables:
##  $ chromosome            : chr  &quot;10&quot; &quot;10&quot; &quot;10&quot; &quot;10&quot; ...
##  $ start                 : int  68574062 68574076 68574087 68574088 68574093 68574101 68574101 68574102 68574106 68574107 ...
##  $ reference             : chr  &quot;C&quot; &quot;C&quot; &quot;G&quot; &quot;G&quot; ...
##  $ alternate             : chr  &quot;T&quot; &quot;A&quot; &quot;C&quot; &quot;C&quot; ...
##  $ hgvs                  :List of 50
##  $ displayConsequenceType: chr  &quot;missense_variant&quot; &quot;missense_variant&quot; &quot;missense_variant&quot; &quot;missense_variant&quot; ...
##  $ consequenceTypes      :List of 50
##  $ conservation          :List of 50
##  $ geneExpression        :List of 50
##  $ geneTraitAssociation  :List of 50
##  $ geneDrugInteraction   :List of 50
##  $ traitAssociation      :List of 50
##  $ cytoband              :List of 50
##  $ id                    : chr  NA NA &quot;rs776152780&quot; NA ...
##  $ populationFrequencies :List of 50
##  $ repeat                :List of 50
</code></pre>

<h1 id="cellbaser-wrappers">cellbaseR wrappers</h1>

<p>cellbaseE package also provides many wrapper functions around commomn cellbaseR
queries. These include:<br />
- getClinicalByGene
- getTranscriptByGene
- getGeneInfo
- getSnpByGene
- getProteinInfo
- getClinicalByRegion
- getConservationByRegion
- getRegulatoryByRegion
- getTfbsByRegion
- getCaddScores
- getVariantAnnotation</p>

<h1 id="cellbaser-utilities">CellbaseR utilities</h1>

<h2 id="creategenemodel">CreateGeneModel</h2>

<p>A usefull utility for fast building of gene models, compatible with Gviz
package for genomic visualization</p>

<pre><code class="language-r">library(cellbaseR)
cb &lt;- CellBaseR()
test &lt;- createGeneModel(object = cb, region = &quot;17:1500000-1550000&quot;)
if(require(&quot;Gviz&quot;)){
  testTrack &lt;- Gviz::GeneRegionTrack(test)
  Gviz::plotTracks(testTrack, transcriptAnnotation='symbol')
}
</code></pre>

<p><img src="/post/2019-10-02-simplifying-genomic-annotations-in-r_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>

<h2 id="annotatevcf">AnnotateVcf</h2>

<p>This utility allows users to annoate genomic variants from small to medium-sized
vcf files directly from the cellbase server, with a wealth of genomic
annotations including riche clinical annotations like clinvar, gwas, and cosmic
data, as well as conservation and functional scores like phast and cadd  scores
, without the need to download any files.</p>

<pre><code class="language-r">library(cellbaseR)
library(VariantAnnotation)
cb &lt;- CellBaseR()
fl &lt;- system.file(&quot;extdata&quot;, &quot;hapmap_exome_chr22_200.vcf.gz&quot;,package = &quot;cellbaseR&quot; )
res &lt;- AnnotateVcf(object=cb, file=fl, BPPARAM = bpparam(workers=2),batch_size = 100)
vcf &lt;- readVcf(fl, &quot;hg19&quot;)
samples(header(vcf))
</code></pre>

<pre><code>##  [1] &quot;NA07034@1099927558&quot; &quot;NA07048@1099927687&quot; &quot;NA07055@1099927615&quot;
##  [4] &quot;NA10846@1099927836&quot; &quot;NA10847@1099927741&quot; &quot;NA12146@1099927743&quot;
##  [7] &quot;NA12239@1099927424&quot; &quot;NA12877@1099925716&quot; &quot;NA12878@1099927697&quot;
## [10] &quot;NA12891@1099927856&quot; &quot;NA12892@1099927810&quot; &quot;NA18503@1099927775&quot;
## [13] &quot;NA18504@0178873056&quot; &quot;NA18505@1099927412&quot; &quot;NA18506@1099927650&quot;
## [16] &quot;NA18524@1099927756&quot; &quot;NA18532@1099927601&quot; &quot;NA18912@1099927835&quot;
## [19] &quot;NA18913@1099927630&quot; &quot;NA18914@0178874379&quot; &quot;NA18940@1099927867&quot;
## [22] &quot;NA18947@0178875080&quot;
</code></pre>

<pre><code class="language-r">length(rowRanges(vcf))==nrow(res)
</code></pre>

<pre><code>## [1] TRUE
</code></pre>

<pre><code class="language-r">str(res,1)
</code></pre>

<pre><code>## 'data.frame':	200 obs. of  18 variables:
##  $ chromosome             : chr  &quot;22&quot; &quot;22&quot; &quot;22&quot; &quot;22&quot; ...
##  $ start                  : int  16157603 17060707 17072347 17177682 17265124 17326914 17446991 17469049 17602839 17603503 ...
##  $ reference              : chr  &quot;G&quot; &quot;G&quot; &quot;C&quot; &quot;C&quot; ...
##  $ alternate              : chr  &quot;C&quot; &quot;A&quot; &quot;T&quot; &quot;A&quot; ...
##  $ id                     : chr  &quot;rs370790235&quot; &quot;rs117836313&quot; &quot;rs139948519&quot; &quot;rs3020747&quot; ...
##  $ hgvs                   :List of 200
##  $ displayConsequenceType : chr  &quot;intron_variant&quot; &quot;2KB_downstream_variant&quot; &quot;stop_gained&quot; &quot;non_coding_transcript_exon_variant&quot; ...
##  $ consequenceTypes       :List of 200
##  $ populationFrequencies  :List of 200
##  $ conservation           :List of 200
##  $ geneExpression         :List of 200
##  $ geneTraitAssociation   :List of 200
##  $ geneDrugInteraction    :List of 200
##  $ functionalScore        :List of 200
##  $ cytoband               :List of 200
##  $ repeat                 :List of 200
##  $ variantTraitAssociation:'data.frame':	200 obs. of  2 variables:
##  $ traitAssociation       :List of 200
</code></pre>

    </div>

    


    

<div class="article-tags">
  
  <a class="badge badge-light" href="/tags/genomics/">Genomics</a>
  
  <a class="badge badge-light" href="/tags/cellbaser/">cellbaseR</a>
  
  <a class="badge badge-light" href="/tags/variant-annotation/">variant Annotation</a>
  
  <a class="badge badge-light" href="/tags/bioinformatics/">Bioinformatics</a>
  
  <a class="badge badge-light" href="/tags/opencga/">OpenCGA</a>
  
</div>



    
      








  






  
  
  
    
  
  
  <div class="media author-card">
    
      
      <img class="portrait mr-3" src="/authors/melsiddieg/avatar_hu08676096bfc58e7bfdd9eae9c4b6bf9f_415077_250x250_fill_q90_lanczos_center.jpg" alt="Avatar">
    

    <div class="media-body">
      <h5 class="card-title"><a href="https://melsiddieg.github.io/">Mohammed O.E Abdallah</a></h5>
      <h6 class="card-subtitle">Post-Doctoral Fellow, Department of Human Genetics</h6>
      <p class="card-text">My research interests include genomics data science, rare genetics disoders and scientific computing.</p>
      <ul class="network-icon" aria-hidden="true">
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="/#contact" >
        <i class="fas fa-envelope"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="https://twitter.com/mohammedsiddieg" target="_blank" rel="noopener">
        <i class="fab fa-twitter"></i>
      </a>
    </li>
  
    
    
    
    
    
    
    
      
    
    <li>
      <a href="https://scholar.google.com/citations?user=WJNLulIAAAAJ&amp;hl=en" target="_blank" rel="noopener">
        <i class="ai ai-google-scholar"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="https://github.com/melsiddieg" target="_blank" rel="noopener">
        <i class="fab fa-github"></i>
      </a>
    </li>
  
</ul>

    </div>
  </div>



      
      
    

    

    


  </div>
</article>

      

    
    

    
    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js" integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js" integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" integrity="sha256-X5PoE3KU5l+JcX+w09p/wHl9AzK333C4hJ2I9S5mD4M=" crossorigin="anonymous"></script>

      

      
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/highlight.min.js" integrity="sha256-aYTdUrn6Ow1DDgh5JTc3aDGnnju48y/1c8s1dgkYPQ8=" crossorigin="anonymous"></script>
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/languages/r.min.js"></script>
        
      

      
      
    

    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.2.0/leaflet.js" integrity="sha512-lInM/apFSqyy1o6s89K4iQUKg6ppXEgsVxT35HbzUupEVRh2Eu9Wdl4tHj7dZO0s1uvplcYGmt3498TtHq+log==" crossorigin="anonymous"></script>
    

    
    
    <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script>
      const search_index_filename = "/index.json";
      const i18n = {
        'placeholder': "Search...",
        'results': "results found",
        'no_results': "No results found"
      };
      const content_type = {
        'post': "Posts",
        'project': "Projects",
        'publication' : "Publications",
        'talk' : "Talks"
        };
    </script>
    

    
    

    
    
    <script id="search-hit-fuse-template" type="text/x-template">
      <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
      </div>
    </script>
    

    
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.2.1/fuse.min.js" integrity="sha256-VzgmKYmhsGNNN4Ph1kMW+BjoYJM2jV5i4IlFoeZA9XI=" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>
    

    
    

    
    

    
    
    
    
    
    
    
    
    
      
    
    
    
    
    <script src="/js/academic.min.0bf1e3db85cbb232372ed31d6f10dc70.js"></script>

    






  
  <div class="container">
    <footer class="site-footer">
  

  <p class="powered-by">
    

    Powered by the
    <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
    <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

    
    <span class="float-right" aria-hidden="true">
      <a href="#" class="back-to-top">
        <span class="button_icon">
          <i class="fas fa-chevron-up fa-2x"></i>
        </span>
      </a>
    </span>
    
  </p>
</footer>

  </div>
  

  
<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Cite</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        <pre><code class="tex hljs"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-outline-primary my-1 js-copy-cite" href="#" target="_blank">
          <i class="fas fa-copy"></i> Copy
        </a>
        <a class="btn btn-outline-primary my-1 js-download-cite" href="#" target="_blank">
          <i class="fas fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

</body>
</html>
